File: HomeTask2/HomeTask2/Computer.cs
using System;

namespace HomeTask2
{
    public abstract class Computer
    {
        private string _macAddress;
        private string _name;

        public string MacAddress
        {
            get { return _macAddress; }
            set
            {
                _macAddress = value;
            }
        }

        public string Name { get => _name; set => _name = value; }

        public Computer(string name, string macAddress)
        {
            MacAddress = macAddress;
        }
        public Computer()
        {

        }

        public abstract void TurnOn();
        public abstract void TurnOff();
    }
}

File: HomeTask2Kotlin/src/Computer.kt
public abstract class Computer(var macAddress:String) {

    //    var macAddress: String = ""
//        set(value)
//        { field = value}
//
//    constructor(){}
//    constructor(macAddress: String) : this(){
//        this.macAddress = macAddress
//    }
    abstract fun turnOn();
    abstract fun turnOff();

    override fun toString(): String {
        return String.format("macAddress = ${macAddress} ")
    }
}